### DISCUSSION TITLE
Issues installing the Dev Environment

### DISCUSSION THREAD

**Discussion by werberger** [CiviWiki/OpenCiviWiki#1305] [Help and support]
Hi all,

New to contribution to open source projects. New to code.

So please let me know if there is a more appropriate way to seek support with questions like this.

I have been assigned my first issue (#127 ) and in trying to set up my environment, I believe I've install poetry correctly, and created my local branch. Following the [instructions](https://github.com/CiviWiki/OpenCiviWiki/blob/develop/CONTRIBUTING.md#development) I am stuck at running 'collectstatic'. I believe the issue is because the project is running Django = "^4.1" (according to the pyproject.toml) based on my reading [here](https://github.com/encode/django-rest-framework/pull/8591).

Assuming my understanding is correct, am I good to downgrade my Django for the purposes of working on my branch or is this an issue with the project?

If I am wrong in my understanding, any suggestions?

Again, please let me know how I should seek help like this if not here.

Cheers.

**Comment by brylie** [comment]
What is the error you see?

**Comment by werberger** [comment]
the error is:

Post-processing 'rest_framework\css\bootstrap-theme.min.css' failed!

I downgraded django to test and it allowed me to run collectstatic.

However, when I run the server I end up with only the navbar and footer on the pages. Not sure if this is another issue or one resulting from having downgraded django.

**Reply by brylie** [reply]
The project is a heavy construction zone. One issue underway is to remove the REST framework so we can holistically focus on core project features.

Once we remove the rest framework, we will build standard Django views for CRUD operations related to Civis. We will also hopefully steer towards using more conventional names for things, like Wiki (or tooic) insteas of "Civi."

**Reply by brylie** [reply]
I've fixed a few issues today, including the error you were getting. Try again using the latest `develop` branch.

**Reply by brylie** [reply]
@werberger, can you get the dev environment running with the latest  `develop` branch?

**Reply by werberger** [reply]
Hey Brylie, appreciate that someone else has been assigned the issue I was theoretically working on - fair enough I've been pretty busy recently and not responded to you.

I'm keen to pick up this or another issue, however I'm struggling to get the most recent development branch running. Failing to run the migrations in the virtual env. (Error is "ModuleNotFoundError: No module named 'rich'")

I fear I'm in need of too much support so feel free to say "Thanks, but no thanks". Otherwise, would appreciate some help.

Cheers

**Reply by brylie** [reply]
We've recently fixed several issues and updated our [contributing guide](https://github.com/CiviWiki/OpenCiviWiki/blob/develop/CONTRIBUTING.md). Please follow the [most recent development instructions](https://github.com/CiviWiki/OpenCiviWiki/blob/develop/CONTRIBUTING.md#development) using the Poetry package manager.

Here is an example video showing the commands up to and including running the initial migrations:

[Peek 2022-11-01 20-37.webm](https://user-images.githubusercontent.com/17307/199312148-97cf0a1b-8f8b-4937-b70e-4ee55db0f461.webm)

**Reply by werberger** [reply]
Thanks, I was using it, but think I may have missed a step.

Now I've got it running but getting the following error when running the commands to populate initial data:

> threads.Civi.linked_civis: (fields.W345) related_name has no effect on ManyToManyField with a symmetrical relationship, e.g. to "self".

I have found it mentioned as an issue for the project [here](https://devscope.io/code/CiviWiki/OpenCiviWiki/issues/1427)... do you know where it's at or if there's a work around?

**Reply by brylie** [reply]
> threads.Civi.linked_civis: (fields.W345) related_name has no effect on ManyToManyField with a symmetrical relationship, e.g. to "self".

That's just a warning, so feel free to ignore it.

**Reply by Nehemiah60** [reply]
I have also been facing some problems especially in running the project. For instance, when I try poetry install....the response is no dependencies to install. This I think might have a significant impact on running the project. It runs but the display is not really as intended.
![Screenshot (412)](https://user-images.githubusercontent.com/87471613/215052800-cfae3b34-ee13-4a63-a22a-ea651307e2e2.png)

**Reply by brylie** [reply]
From what directory are you running `poetry install`? Can you share a screen recording of the steps you take prior to running `poetry install`, such as ensuring the correct directory?
